### DISCUSSION TITLE
Is it possible to have Stash rename the files and move files into more organised files and folders?

### DISCUSSION THREAD

**Discussion by GronsoBitburg** [stashapp/stash#2963] [Q&A] [answered]
Is it possible to have Stash rename the files and move files into more organised files?

I was hoping that by finding the metadata for a specific scene and movie that I could organise these automatically into movie folders and scenes, or at the very least rename the files to be better formatted. My Stash database is impecible, I just need a way to clean up the naming conventions of the files and folders, as my sources of the scenes are all over the place, so every file is poorly named or not named at all.

Is this something that is in the works or am I missing this option somewhere?

**Comment by Tweeticoats** [comment, marked as answer]
There are 3rd party plugins that can do this take a look at rename on update https://github.com/stashapp/CommunityScripts.
Depending on if you are running the stable version of stash or running the development version of stash you may need to run a pre-release version of the plugin.
Ask on discord in the plugins and scripts room to ask about plugins.

**Comment by ghost** [comment]
I made [a script](https://github.com/stashapp/CommunityScripts/issues/65) that uses TMSU and Supertag to create a virtual filesystem with the performers, studios and tags, so files are ordered into folders with symlinks.

There are open issues about this here: [#1658](https://github.com/stashapp/stash/issues/1658), [#2693](https://github.com/stashapp/stash/issues/2693)

**Comment by GronsoBitburg** [comment]
Thank you for your suggestions so far, but I was hoping for a more integrated option, as I am using Docker for my implementation of Stash.

The 'Stash Sqlite Renamer' available on the Community page is specifically for Windows 10. I am running a container on my local Linux server and it's a headache trying to get plugins to play nice with containers. Creating a further virtual filesystem using TMSU and Supertag also seems like a pain to set up.

My question was more aimed at whether there is any development on the horizon utilising the use of the Organise button to help rename and move files into a more structured filing system. Not necessarily plugins, which have the habit to not be adequately maintained, unlike the Stash app.

**Reply by ghost** [reply]
> My question was more aimed at whether there is any development on the horizon utilising the use of the Organise button to help rename and move files into a more structured filing system.

The issue I linked previously is asking for just that but it hasn't been assigned to any Milestone yet so don't hold your breath. There is no development planned on the horizon.

**Comment by 7dJx1qP** [comment]
> The 'Stash Sqlite Renamer' available on the Community page is specifically for Windows 10. I am running a container on my local Linux server and it's a headache trying to get plugins to play nice with containers. Creating a further virtual filesystem using TMSU and Supertag also seems like a pain to set up.

Have you checked the renamerOnUpdate plugin? That's the one being actively worked on and what I see everyone using. I believe there's people using it on various platforms and not just Windows. https://github.com/stashapp/CommunityScripts/tree/main/plugins/renamerOnUpdate

> My question was more aimed at whether there is any development on the horizon utilising the use of the Organise button to help rename and move files into a more structured filing system. Not necessarily plugins, which have the habit to not be adequately maintained, unlike the Stash app.

A feature for renaming files is in the backlog. https://github.com/stashapp/stash/issues/884

**Comment by Br0kenx** [comment]


**Reply by ghost** [reply]
Do you know how to program in golang?

https://lemmy.ml/post/546204
